gpu_time/gpu_time_test.cpp

```diff
@@ -0,0 +1,434 @@
+    timestamps[7] = 1;           // Availability
+
+    // Simulate a 30ms duration for the third command buffer (queries 4 and 5)
+    // 30ms = 30,000,000 ns
+    timestamps[8] = 3000000000;   // Start time for cmd 3
+    timestamps[9] = 1;            // Availability
+    timestamps[10] = 3030000000;  // End time for cmd 3
+    timestamps[11] = 1;           // Availability
+
+    return VK_SUCCESS;
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL MockQueueWaitIdle(VkQueue queue)
+{
+    // No-op for testing
+    return VK_SUCCESS;
+}
+
+VKAPI_ATTR VkResult VKAPI_CALL MockDeviceWaitIdle(VkDevice device)
+{
+    // No-op for testing
+    return VK_SUCCESS;
+}
+
+void CreateGPUTime(Dive::GPUTime& gpuTime, float timestamp_period)
+{
+    ASSERT_TRUE(gpuTime
+                .OnCreateDevice(MOCK_DEVICE,
+                                /*allocator=*/nullptr,
+                                timestamp_period,
+                                MockCreateQueryPool,
+                                MockResetQueryPool)
+                .success);
+}
+
+void DestroyGPUTime(Dive::GPUTime& gpuTime)
+{
+    VkQueue queue = MOCK_QUEUE;
+    gpuTime.OnGetDeviceQueue(&queue);
+    ASSERT_TRUE(
+    gpuTime.OnDestroyDevice(MOCK_DEVICE, MockQueueWaitIdle, MockDestroyQueryPool).success);
+}
+
+void ExpectStatsEq(const Dive::GPUTime::Stats& stats,
+                   double                      expected_average,
+                   double                      expected_median,
+                   double                      expected_min,
+                   double                      expected_max,
+                   double                      expected_stddev)
+{
+    EXPECT_DOUBLE_EQ(stats.average, expected_average);
+    EXPECT_DOUBLE_EQ(stats.median, expected_median);
+    EXPECT_DOUBLE_EQ(stats.min, expected_min);
+    EXPECT_DOUBLE_EQ(stats.max, expected_max);
+    EXPECT_DOUBLE_EQ(stats.stddev, expected_stddev);
+}
+
+// --- GPUTime Tests ---
+// Test that a newly created object returns default statistics.
+TEST(GPUTimeTest, InitialStateReturnsDefaultStats)
+{
+    Dive::GPUTime gpuTime;
+    CreateGPUTime(gpuTime, kMockTimestampPeriod);
+
+    auto stats = gpuTime.GetStats();
+    ExpectStatsEq(stats,
+                  /* average */ 0.0,
+                  /* median */ 0.0,
+                  /* min */ std::numeric_limits<double>::max(),
+                  /* max */ std::numeric_limits<double>::lowest(),
+                  /* stddev */ 0.0);
+
+    DestroyGPUTime(gpuTime);
+}
+
+// Test that a command buffer can be successfully allocated.
+TEST(GPUTimeTest, AllocateCommandBufferSucceeds)
+{
+    Dive::GPUTime gpuTime;
+    CreateGPUTime(gpuTime, kMockTimestampPeriod);
+
+    VkCommandBufferAllocateInfo allocInfo = {};
+    allocInfo.commandPool = MOCK_COMMAND_POOL;
+    allocInfo.commandBufferCount = 1;
+    VkCommandBuffer cmdBuf = MOCK_COMMAND_BUFFER_1;
+
+    auto status = gpuTime.OnAllocateCommandBuffers(&allocInfo, &cmdBuf);
+    ASSERT_TRUE(status.success);
```

Comment:
Consider inlining `status` so that the test output includes more information. This is widely applicable to the PR.

```suggestion
    ASSERT_TRUE(gpuTime.OnAllocateCommandBuffers(&allocInfo, &cmdBuf).success);
```

Reply:
fixed